webapp/packages/core-root/src/SessionPermissionsResource.ts

```diff
@@ -7,20 +7,47 @@
  */
 import { injectable } from '@cloudbeaver/core-di';
 import { ExecutorInterrupter } from '@cloudbeaver/core-executor';
+import { LocalizationService } from '@cloudbeaver/core-localization';
 import { CachedDataResource, CachedResource, GraphQLService } from '@cloudbeaver/core-sdk';
 
+import { DataSynchronizationService } from './DataSynchronization/DataSynchronizationService';
 import { SessionDataResource } from './SessionDataResource';
+import { ServerEventId } from './SessionEventSource';
+import { ISessionPermissionEvent, SessionPermissionEventHandler } from './SessionPermissionEventHandler';
+import { SessionResource } from './SessionResource';
 
 @injectable()
 export class SessionPermissionsResource extends CachedDataResource<Set<string>> {
-  constructor(private readonly graphQLService: GraphQLService, sessionDataResource: SessionDataResource) {
+  constructor(
+    private readonly graphQLService: GraphQLService,
+    private readonly dataSynchronizationService: DataSynchronizationService,
+    private readonly localizationService: LocalizationService,
+    sessionDataResource: SessionDataResource,
+    sessionResource: SessionResource,
+    sessionPermissionEventHandler: SessionPermissionEventHandler,
+  ) {
     super(() => new Set());
 
     this.sync(
       sessionDataResource,
       () => {},
       () => {},
     );
+
+    sessionPermissionEventHandler.onEvent<ISessionPermissionEvent>(
+      ServerEventId.CbSubjectPermissionsUpdated,
+      () => {
+        this.dataSynchronizationService
+          .requestSynchronization('permissions', this.localizationService.translate('app_root_event_permissions_changed_message'))
```

Comment:
probably you can't use core-localization here (circular dependency between packages)